core/src/main/java/hudson/WorkspaceSnapshot.java

```diff
@@ -30,20 +30,9 @@
 import java.io.IOException;
 
 /**
- * Represents a workspace snapshot created by {@link FileSystemProvisioner}.
- *
- * <p>
- * This class encapsulates a logic to use the snapshot elsewhere.
- * The instance will be persisted with the {@link AbstractBuild} object
- * as an {@link Action}.
- *
- * <p>
- * TODO: how to garbage-collect this object, especially for zfs?
- * perhaps when a new build is started?
- *
- * @see FileSystemProvisioner
- * @author Kohsuke Kawaguchi
+ * @deprecated Unused.
```

Comment:
Actually https://plugins.jenkins.io/clone-workspace-scm/ (untouched for seven years, but apparently still installed by many people) _does_ implement this, but without using `FileSystemProvisioner`.